### PR TITLE
feat: implemented api call and displayed offers in list

### PR DESCRIPTION
**Pull request by wawrzonnn**

### CODE REVIEW

**Review by NerdbordMentor — APPROVED** [review]
## Code Review Report

### Task Goal
Fetch job offers from an external API.

### Requirements
1. API should be called on the specified endpoint.
2. Job offers should be displayed in a list.
3. React Query library must be used.

### Code Changes Review

#### package.json
The `react-query` and `date-fns` libraries have been added to the dependencies. This is in line with the task requirements.

#### src/App.tsx
The `QueryClientProvider` from `react-query` has been added and wrapped around the main application. This is necessary for using React Query in the application and is in line with the task requirements.

#### src/api/jobOffers.ts
A new file has been added to handle fetching job offers from the API. The specified endpoint is used, which meets the first requirement.

#### src/api/queryClient.ts
A new file has been added to create a new instance of `QueryClient` from `react-query`. This is necessary for using React Query in the application and is in line with the task requirements.

#### src/api/types.ts
A new file has been added to define the types related to job offers. This is good for type safety and readability.

#### src/components/OffersList/OffersList.tsx
The `useQuery` hook from `react-query` is used to fetch job offers, which meets the third requirement. The fetched job offers are displayed in a list, which meets the second requirement.

In conclusion, all the code changes are in line with the task requirements. There are no apparent bugs in the code changes.
